### PR TITLE
AWS Secret Manager for DB credentials

### LINKED ISSUE #126: Use a vault for DB credentials
Today, the `pkg/storage/database` package is responsible for fetching the credentials for a database. That interface contains a function called [GetDestinationCredentials()](https://github.com/scratchdata/scratchdata/blob/main/pkg/storage/database/database.go#L17).

We want to refactor the code to use a separate package for managing secrets.

## Step 1: Create `pkg/storage/vault` package

This will follow the same pattern as the `pkg/storage/blobstore` package, where different storage types can be configured.

- [ ] The new `vault` package should have an interface which has two functions: `GetCredential(name string) string` and `SetCredential(name, value string)`
- [ ] There should be a default implementation called `memory` which just returns credentials from `config.Destination` in YAML. The implementation should basically be identical to the existing functionality [here](https://github.com/scratchdata/scratchdata/blob/main/pkg/storage/database/memory/memory.go#L132).
- [ ] There should be a new configuration section in our yaml file called `vault` which has the same format as the `database` section (type and settings.)

## Step 2: Refactor code to use the new `vault` package instead of `database`

- [ ] Delete the [GetDestinationCredentials()](https://github.com/scratchdata/scratchdata/blob/main/pkg/storage/database/database.go#L17) function from the Database interface.
- [ ] Refactor the code to use our new vault instead. This means updating [GetStorageServices()](https://github.com/scratchdata/scratchdata/blob/main/cmd/scratchdata/scratchdata.go#L69) to include a Vault as part of the struct and replacing any use of `GetDestinationCredentials()` to use our new vault instead

## Step 3: AWS Secrets Manager 

Create a new implementation of the Vault interface which uses AWS Secrets manager

- [ ] Create a new package under `vault` for AWS.
- [ ] Use the go-v2 API to implement the vault interface
- [ ] The configuration file should take AWS credentials (access id, secret key) along with a `prefix`. When keys are created or retrieved, we should add this prefix.

### PR DESCRIPTION
**Pull request by mehulmathur16**
/claim #126
/fixes #126

Hi @poundifdef, I have tested out the memory vault & it is working as expected. In case of aws vault, I do not have an aws account, so wasn't able to test it out. Let me know if you face any issues.

Additionally, I have added Region to the aws config.

### CODE REVIEW

**Comment by poundifdef** [comment]
Can you create an AWS account and test it? If you can confirm that it works then I’m happy to review.

File: pkg/destinations/destinations.go [hunk 1]
```diff
@@ -56,13 +66,21 @@ func (m *DestinationManager) Destination(databaseID int64) (Destination, error)
...
 		var dest Destination
+		var creds Creds
 
 		dest, ok := m.pool[databaseID]
 		if ok {
 			return dest, nil
 		}
 
-		creds, err := m.storage.Database.GetDestinationCredentials(databaseID)
+		jsonCreds, err := m.storage.Vault.GetCredential(strconv.Itoa(int(databaseID)))
```
Comment (mehulmathur16) [inline pkg/destinations/destinations.go:76]: @poundifdef in which format should the credentials be stored on aws secrets manager ?
Also, should this logic be different when reading from different vaults. As the memory vault returns a json string consisting of all destination details, which is here decoded to Credentials type.

**Comment by mehulmathur16** [comment]
Sure 👍

**Comment by mehulmathur16** [comment]
@poundifdef This branch is too old from main. So creating PR from different branch & closing this.

**Pull request closed without merging**